src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -526,7 +526,18 @@ def get(self, request: HttpRequest, organization: Organization, trace_id: str) -
         except NoProjects:
             return Response(status=404)
 
+        trace_view_load_more_enabled = features.has(
+            "organizations:trace-view-load-more",
+            organization,
+            actor=request.user,
+        )
+
         detailed: bool = request.GET.get("detailed", "0") == "1"
+        limit: int = (
+            request.GET.get("limit", MAX_TRACE_SIZE)
+            if trace_view_load_more_enabled
+            else MAX_TRACE_SIZE
+        )
```

Comment:
you might wanna have an upper bound for `limit` as well in case a ridiculously high number was passed in

Reply:
Yep, agree!